src/helmRepoIndex/main.go

```diff
@@ -0,0 +1,135 @@
+/********************************************************************************
+ * Copyright (c) 2021,2022 Mercedes-Benz Tech Innovation Gmbh
+ * Copyright (c) 2021,2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/go-github/v48/github"
+	"golang.org/x/oauth2"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+func main() {
+	var centralHelmIndex string
+	var gitOwner string
+	var gitToken string
+
+	flag.StringVar(&centralHelmIndex, "indexFile", "", "Specify Helm Repository index file")
+	flag.StringVar(&gitOwner, "owner", "", "Specify GitHub User or Organization")
+	flag.StringVar(&gitToken, "token", "", "Specify GitHub Token")
+	flag.Parse()
+
+	ctx := context.Background()
+	client := getAuthenticatedClient(ctx, gitToken)
+	repos, _, _ := getOrgRepos(ctx, gitOwner, client)
+
+	for i, gitRepo := range repos {
+		var gitRepoHelmIndex string
+
+		//check gh pages configured for repo
+		_, response, _ := client.Repositories.GetBranch(ctx, gitOwner, *gitRepo.Name, "gh-pages", false)
+
+		if response.StatusCode == 200 {
+			fmt.Println(i, *gitRepo.Name)
+			gitRepoHelmIndex = downloadProductHelmRepoIndex(ctx, client, gitOwner, *gitRepo.Name)
+			buildHelmRepoIndex(centralHelmIndex, gitRepoHelmIndex)
+		}
+	}
+}
+
+func getAuthenticatedClient(ctx context.Context, gitToken string) *github.Client {
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: gitToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	return github.NewClient(tc)
+}
+
+func getOrgRepos(ctx context.Context, gitOwner string, client *github.Client) ([]*github.Repository, *github.Response, error) {
+	opt := &github.RepositoryListByOrgOptions{
+		Type: "all",
+	}
+	repos, response, err := client.Repositories.ListByOrg(ctx, gitOwner, opt)
```

Comment:
Rico used a github client too and had to verify that there are no 'next' results. is this caught here somehow?

Reply:
As `PerPage` is not explicit defined for `github.ListOptions` (which defaults to 0 to my understanding), I expect no issues here. [Package documentation](https://pkg.go.dev/github.com/google/go-github/v48@v48.1.0/github#hdr-Pagination) also state, that it is supported. I don't derive a _you must conifigure it_ from doc.

Reply:
i'm not concerned that we need to but that the amount of results you get is limited if you don't fetch the rest of it.